### ISSUE TITLE
Hassio HomeAssistant 0.54 & 0.55 - Issue Cannot Install HABridge Add-On (Resolved !!!)

### ISSUE THREAD

**Issue by ibexcentral** [notoriousbdg/hassio-addons#8]
Hi,

I am running Hassio 1.1 but also tried on 1.0 same issue, I am running HomeAssistant 0.54:

https://github.com/notoriousbdg/hassio-addons

When I try to install Ha Bridge Add-on on a fresh install of Hassio with HomeAssistant v0.54 on RaspberryPi3, the Install button turns red immediately the Supervisor log as follows:

```
17-10-02 08:52:15 INFO (MainThread) [hassio.addons.addon] Create Home-Assistant addon data folder /data/addons/data/core_ssh
17-10-02 08:52:15 INFO (SyncWorker_1) [hassio.dock.interface] Pull image homeassistant/armhf-addon-ssh tag 2.8.
17-10-02 08:53:13 INFO (SyncWorker_1) [hassio.dock.interface] Tag image homeassistant/armhf-addon-ssh with version 2.8 as latest
17-10-02 08:53:38 INFO (SyncWorker_17) [hassio.dock.addon] Start docker addon homeassistant/armhf-addon-ssh with version 2.8
17-10-02 08:54:22 INFO (MainThread) [hassio.addons.git] Clone addon https://github.com/notoriousbdg/hassio-addons repository
17-10-02 08:54:24 INFO (MainThread) [hassio.addons] Load addons: 18 all - 3 new - 0 remove
17-10-02 08:54:29 INFO (MainThread) [hassio.addons.addon] Create Home-Assistant addon data folder /data/addons/data/27e642c6_ha-bridge
17-10-02 08:54:29 INFO (SyncWorker_2) [hassio.dock.addon] Start build 27e642c6/armhf-addon-ha-bridge:4.5.6p2
17-10-02 08:54:29 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 422, in start
    resp = yield from self._request_handler(request)
  File "/usr/lib/python3.6/site-packages/aiohttp/web.py", line 306, in _handle
    resp = yield from handler(request)
  File "/usr/lib/python3.6/site-packages/hassio/api/util.py", line 31, in wrap_api
    answer = await method(api, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/hassio/api/addons.py", line 171, in install
    addon.install(version=version), loop=self.loop)
  File "/usr/lib/python3.6/site-packages/hassio/addons/addon.py", line 443, in install
    if not await self.docker.install(version):
  File "/usr/lib/python3.6/site-packages/hassio/dock/util.py", line 18, in wrap_api
    return await method(api, *args, **kwargs)
  File "/usr/lib/python3.6/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/lib/python3.6/site-packages/hassio/dock/addon.py", line 196, in _install
    return self._build(tag)
  File "/usr/lib/python3.6/site-packages/hassio/dock/addon.py", line 209, in _build
    image = self.docker.images.build(**build_env.get_docker_args(tag))
  File "/usr/lib/python3.6/site-packages/docker/models/images.py", line 175, in build
    raise BuildError(chunk['error'])
docker.errors.BuildError: Please provide a source image with `from` prior to run
````

**Comment by adrian-ludwig** [comment]
All addons are the same.

**Comment by ibexcentral** [comment]
Hi...are you saying that all third party add ons are broken? As the hassio standard addons are installing okay. What are the next steps? Is it possible to downgrade hassio homeassitant to 0.53 whilst being fixed?

**Comment by pcping** [comment]
the same

**Comment by pickeld** [comment]
please fix this

**Comment by pvizeli** [comment]
Need a `build.json` like: https://github.com/home-assistant/hassio-addons/tree/master/google_assistant

**Comment by ravensteijn** [comment]
Did you get it running pvizeli when you added the build.json? In other words, does it work now?

**Comment by ravensteijn** [comment]
I just added build.json and changed the first 2 lines in the docker file. I can't test it now. Is that all?

**Comment by pvizeli** [comment]
Yes that's all :+1:

**Comment by d3nnis0** [comment]
I can confirm with the build.json and modifying the first two lines of the docker file the installation is working again.

**Comment by ravensteijn** [comment]
Hum. I changed it in my fork repo. I stil got errors. Which repo did you use d3nnis0?

**Comment by d3nnis0** [comment]
Hi ravenstein, I installed in local by putting the files in the addon folder... Just checked your repo, seems your build.json is wrong. Paste the following in your Build.json:

{
    "build_from": {
      "armhf": "nodered/node-red-docker:rpi",
      "amd64": "nodered/node-red-docker:latest"  
    }
  }

Let me know if it works :-)

**Comment by ravensteijn** [comment]
D3nnis0,

Thank you for your respons.

It got installed but I get this error. And HA bridge also isn't working. Thank you for your help. 

nodered log
`/run.sh: line 26: npm: command not found
`
in the supervisor log 
```
17-10-09 17:05:50 INFO (SyncWorker_7) [hassio.dock.addon] Start docker addon bfb9ad38/armhf-addon-nodered with version 0.1.2
17-10-09 17:05:50 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 422, in start
    resp = yield from self._request_handler(request)
  File "/usr/lib/python3.6/site-packages/aiohttp/web.py", line 306, in _handle
    resp = yield from handler(request)
  File "/usr/lib/python3.6/site-packages/hassio/api/util.py", line 31, in wrap_api
    answer = await method(api, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/hassio/api/addons.py", line 132, in info
    ATTR_WEBUI: addon.webui,
  File "/usr/lib/python3.6/site-packages/hassio/addons/addon.py", line 222, in webui
    port = self.ports.get("{}/tcp".format(t_port), t_port)
AttributeError: 'NoneType' object has no attribute 'get'

```

**Comment by d3nnis0** [comment]
just compared your repo against mine, looks the same now. Please make sure you uninstall node red and install again from your repo. the message /run.sh: line 26: npm: command not found still looks like the old build.json file...

**Comment by ravensteijn** [comment]
Yes! NodeRED is working! thank you for your help D3nnisO ! 
Next step is HA bridge :-)

i get this error
`docker.errors.BuildError: pull access denied for ha-bridge/ha-bridge-docker, repository does not exist or may require 'docker login'`

**Comment by ravensteijn** [comment]
ha-bridge is working now too!  Thank you all for your help. if i need to commit these changes to the master branch i like to hear. now it's time to configure :+1:

**Comment by ibexcentral** [comment]
Hi, which repo do I use because https://github.com/notoriousbdg/hassio-addons not working for HABridge or NodeRED

**Comment by ravensteijn** [comment]
My repo is working with hassio now

**Comment by ibexcentral** [comment]
Hi thanks what is your repo? and will it work on the raspberry pi3?

**Comment by pi3rr3hassio** [comment]
Hi @ravensteijn, I installed the add-on from your repo, thank you everyone to making the installation working again. But when i tried to start ha-bridge, i got the red button, how did you do to make it works ? i am on Hassio with a pi3

**Comment by ravensteijn** [comment]
https://github.com/ravensteijn/hassio-addons
I'm on raspberry3 and hass.io

What does the supervisor log say?

**Comment by pi3rr3hassio** [comment]
i tried to reinstall it but still on the red button when i start it. see below the log. `17-10-09 21:36:56 INFO (MainThread) [hassio.snapshots] Found 5 snapshot files
17-10-09 21:36:56 INFO (MainThread) [hassio.addons.git] Pull addon https://github.com/notoriousbdg/hassio-addons repository
17-10-09 21:36:56 INFO (MainThread) [hassio.addons.git] Pull addon https://github.com/home-assistant/hassio-addons repository
17-10-09 21:36:56 INFO (MainThread) [hassio.host_control] Send 'info' to HostControl.
17-10-09 21:36:56 ERROR (MainThread) [hassio.snapshots.snapshot] Can't read snapshot tarfile /data/backup/._f27f1df6.tar -> invalid header
17-10-09 21:36:57 INFO (MainThread) [hassio.host_control] Receive from HostControl: { "type": "resinos", "version": "1.1", "last_version": "1.1", "hostname": "hassio", "features": ["shutdown", "reboot", "update", "hostname"], "os": "GNU/Linux" }.
17-10-09 21:36:57 INFO (MainThread) [hassio.addons] Load addons: 24 all - 0 new - 0 remove
17-10-09 21:38:34 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 422, in start
    resp = yield from self._request_handler(request)
  File "/usr/lib/python3.6/site-packages/aiohttp/web.py", line 306, in _handle
    resp = yield from handler(request)
  File "/usr/lib/python3.6/site-packages/hassio/api/util.py", line 31, in wrap_api
    answer = await method(api, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/hassio/api/addons.py", line 132, in info
    ATTR_WEBUI: addon.webui,
  File "/usr/lib/python3.6/site-packages/hassio/addons/addon.py", line 222, in webui
    port = self.ports.get("{}/tcp".format(t_port), t_port)
AttributeError: 'NoneType' object has no attribute 'get'
17-10-09 21:38:35 INFO (SyncWorker_7) [hassio.dock.interface] Remove docker local/armhf-addon-ha-bridge with latest and 4.5.6p2
17-10-09 21:38:35 INFO (MainThread) [hassio.addons.addon] Remove Home-Assistant addon data folder /data/addons/data/local_ha-bridge
17-10-09 21:38:44 INFO (MainThread) [hassio.addons.addon] Create Home-Assistant addon data folder /data/addons/data/local_ha-bridge
17-10-09 21:38:44 INFO (SyncWorker_9) [hassio.dock.addon] Start build local/armhf-addon-ha-bridge:4.5.6p2
17-10-09 21:45:10 INFO (SyncWorker_9) [hassio.dock.addon] Build local/armhf-addon-ha-bridge:4.5.6p2 done
17-10-09 21:45:10 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 422, in start
    resp = yield from self._request_handler(request)
  File "/usr/lib/python3.6/site-packages/aiohttp/web.py", line 306, in _handle
    resp = yield from handler(request)
  File "/usr/lib/python3.6/site-packages/hassio/api/util.py", line 31, in wrap_api
    answer = await method(api, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/hassio/api/addons.py", line 132, in info
    ATTR_WEBUI: addon.webui,
  File "/usr/lib/python3.6/site-packages/hassio/addons/addon.py", line 222, in webui
    port = self.ports.get("{}/tcp".format(t_port), t_port)
AttributeError: 'NoneType' object has no attribute 'get'
17-10-09 21:45:13 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 422, in start
    resp = yield from self._request_handler(request)
  File "/usr/lib/python3.6/site-packages/aiohttp/web.py", line 306, in _handle
    resp = yield from handler(request)
  File "/usr/lib/python3.6/site-packages/hassio/api/util.py", line 31, in wrap_api
    answer = await method(api, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/hassio/api/addons.py", line 132, in info
    ATTR_WEBUI: addon.webui,
  File "/usr/lib/python3.6/site-packages/hassio/addons/addon.py", line 222, in webui
    port = self.ports.get("{}/tcp".format(t_port), t_port)
AttributeError: 'NoneType' object has no attribute 'get'`

**Comment by ravensteijn** [comment]
I got the same error logs and ha-bridge did work. Just try to open yourhassio-ip:80 . I don't know What it is. I think the config? Just open the ha-bridge and you can config inside. It's very responsive on my TOON touchscreen thermostat with hue connection

**Comment by pvizeli** [comment]
Bug is fix with webui on 0.70

**Comment by ibexcentral** [comment]
Thanks both when will fix for webui 0.70 be available? Where do I get it and how do I apply it?. Sorry I am a semi noob.

**Comment by pvizeli** [comment]
Wait 8 hours and he do it self or go to adventage tab and click reload and after that update on supervisor.

**Comment by ibexcentral** [comment]
Great thanks so Supervisor updates Webui?

**Comment by pi3rr3hassio** [comment]
thank you guys! it did the trick. yes if you update the supervisor (hassio > the 3 dots on the top left > advance setting ) by reloading it then click on upload, it should be written 0.70. after that, ha-bridge works. (you still have to write your Hassio ip address in the option of the ha-bridge) 👍

**Comment by ibexcentral** [comment]
Awesome thanks to all you guys, great job!

I have been trying to get HABridge working with other options I had to install HomeAssistant 0.53.1 on Raspbian to get HABridge to work but now I am back to Hassio 1.1, HomeAssistant 0.55 and HABridge 4.5.6p2 !

**Comment by pvizeli** [comment]
The best would be if @notoriousbdg would push a ready to use image to dockerhub. So you can only pull the image withoud need build it on local device.

**Comment by ravensteijn** [comment]
Do you mean something like : https://hub.docker.com/r/jritsema/rpi-ha-bridge/ ?
I've seen this docker. what needs to be changed (build.json, dockerfile)  to use this docker? the node-red part is build like this , am i right?

**Comment by notoriousbdg** [comment]
Sorry for the delay.  I was on vacation.  This should be fixed now with d27f28a64af3b8d6b0a805b9cd43d6e645d9f40b.

**Comment by pvizeli** [comment]
@ravensteijn you need only a `"image": "myrepo/image-name"` into `config.json`. For building you can use our builder tool: https://github.com/home-assistant/hassio-build/tree/master/builder
